### PR TITLE
feat(audits): Add new UI

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

This reworks the Audit UI in the top right to generally rework the visual aspect, and also add a detailed view

**List View**:
<details>

image

</details>

**Detailed View**:
<details>

image

</details>

**No audits:** 
<details>

image

</details>

Fix https://github.com/withastro/astro/issues/10327

## Testing

Tests should pass! We don't test every single interaction between the cards and the highlights, nonetheless we still test that everything shows as expected.

## Docs

N/A. In the future, this will allow for cool long descriptions for every audits, which will need a fair amount of docs work, similar to the work on errors.

### CODE REVIEW

**Comment by Princesseuh** [comment]
!preview audits-ui

File: .changeset/witty-wombats-attend.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": minor
+---
+
+Update UI for dev toolbar audits with new information
```
Comment (sarah11918) [inline .changeset/witty-wombats-attend.md:5]:
```suggestion
Updates the UI for dev toolbar audits with new information
```

**Review by sarah11918 — APPROVED** [review]
The tiniest nit, but makes a review worth it :joy_cat:

File: packages/astro/src/runtime/client/dev-toolbar/ui-library/badge.ts [hunk 2]
```diff
@@ -15,9 +15,11 @@ export class DevToolbarBadge extends HTMLElement {
 
 		if (this.hasAttribute('badge-style'))
 			this.badgeStyle = this.getAttribute('badge-style') as BadgeStyle;
+	}
 
+	connectedCallback() {
 		const classes = [`badge--${this.size}`, `badge--${this.badgeStyle}`];
-		this.shadowRoot.innerHTML = `
+		this.shadowRoot.innerHTML += `
```
Comment (Princesseuh) [inline packages/astro/src/runtime/client/dev-toolbar/ui-library/badge.ts:22]: This change will be removed by https://github.com/withastro/astro/pull/10186, but it has the same effect

File: packages/astro/src/runtime/client/dev-toolbar/toolbar.ts [hunk 3]
```diff
@@ -159,6 +159,7 @@ export class AstroDevToolbar extends HTMLElement {
 				opacity: 0;
 				transition: opacity 0.2s ease-in-out 0s;
 				pointer-events: none;
+				user-select: none;
```
Comment (Princesseuh) [inline packages/astro/src/runtime/client/dev-toolbar/toolbar.ts:162]: Just a small thing I noticed while working on this, it's technically possible for the tooltip on hover to get selected by users somehow.

**Review by natemoo-re — APPROVED** [review]
Consider this design approval on the screenshots you shared! Have not looked at the code, but I'm sure it's #NWTWWHB.

**Comment by bluwy** [comment]
It looks like it's not rendered right on Safari (I'm using 17.4) (works well on Firefox and Chrome):

<details>
<summary>Screenshot</summary>


image

image


</details>

Seems like a fixed width needs to be set on the `astro-dev-toolbar-icon` to prevent it from occupying the flex space, since the inner SVG element has `width: 100%`.

---

Also, would be nice to resolve the eslint warnings related to the dev toolbar 😬 https://github.com/withastro/astro/actions/runs/8193520483/job/22407514922?pr=10217#step:8:26

**Comment by Princesseuh** [comment]
I forgot to test Safari 😔 Will fix! Thank you for checking

**Review by ematipico — APPROVED** [review]
Tested it on a real project with some audits, and the changes seem good to me :)

**Pull request merged**